plugins/inputs/jolokia/jolokia.go

```diff
@@ -252,7 +252,14 @@ func (j *Jolokia) Gather(acc telegraf.Accumulator) error {
 							switch t2 := v.(type) {
 							case map[string]interface{}:
 								for k2, v2 := range t2 {
-									fields[measurement+"_"+k+"_"+k2] = v2
+									switch t3 := v2.(type) {
+									case map[string]interface{}:
+										for k3, v3 := range t3 {
+											fields[measurement+"_"+k+"_"+k2+"_"+k3] = v3
+										}
```

Comment:
Well we're unmarshalling json, so cyclical structures aren't possible, so there's no risk of infinite recursion. You'd have to get thousands of levels deep before it would become an issue.
I guess what I'm saying is that I don't see the need for a depth limit. But the code complexity to add it is pretty minor, so I'm not really opposed to it either.